### PR TITLE
Add Service Manager proxy chart

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

This PR adds the Service Manager proxy chart. 

Now you have an option to install the Service Manager offering in your cluster by adding this component:
```yaml
  - name: service-manager-proxy
    namespace: kyma-system
```

and setting proper overrides:
```yaml
apiVersion: v1
kind: Secret
metadata:
  name: service-manager-proxy-overrides
  namespace: kyma-installer
  labels:
    installer: overrides
    component: service-manager-proxy
    kyma-project.io/installation: ""
data:
  sm.password: "pass"
  sm.user: "username"
  config.sm.url: "service_manager_url"
```

after installing that you will see additional cluster brokers:
```
kubectl get clusterservicebroker
NAME                                                                        URL                                                                                                 STATUS   AGE
sm-auditlog-841ce00c-7519-4b67-b357-94cc1020e7b6                            https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/841ce00c-7519-4b67-b357-94cc1020e7b6   Ready    8m38s
sm-auditlog-management-broker-a8009fe2-c87b-459c-8940-82d12526a8e7          https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/a8009fe2-c87b-459c-8940-82d12526a8e7   Ready    8m38s
sm-business-logging-service-broker-9bb921b5-fbba-4819-8438-8dd804751c7a     https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/9bb921b5-fbba-4819-8438-8dd804751c7a   Ready    8m38s
sm-custom-domain-certificates-broker-837d0b0e-5688-4110-9597-36c01b17da92   https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/837d0b0e-5688-4110-9597-36c01b17da92   Ready    8m38s
sm-destination-service-broker-aba4beec-6cb7-4d47-bee2-cf0a48a7f6f7          https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/aba4beec-6cb7-4d47-bee2-cf0a48a7f6f7   Ready    8m38s
sm-feature-flags-broker-69126bef-ab40-48a9-8ce4-5c4755128a34                https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/69126bef-ab40-48a9-8ce4-5c4755128a34   Ready    8m38s
sm-object-storage-broker-06976c41-4b98-4b93-b209-55566bbd5564               https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/06976c41-4b98-4b93-b209-55566bbd5564   Ready    8m38s
sm-xsuaa-4fb4153d-5fc9-44a8-9f9b-817e5222b354                               https://service-manager.cfapps.us30.hana.ondemand.com/v1/osb/4fb4153d-5fc9-44a8-9f9b-817e5222b354   Ready    8m38s
```
**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/6325

### CODE REVIEW

File: resources/service-manager-proxy/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# service-broker-proxy-k8s
```
Comment (polskikiel) [inline resources/service-manager-proxy/README.md]: we have some guideline I think to have it equal as chart name so in this case: `Service Manager Proxy`

File: resources/service-manager-proxy/values.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    url: http://service-manager.dev.cfdev.sh
+    osb_api_path: /v1/osb
+    request_timeout: 6000ms
+    skip_ssl_validation: false
+  producer:
+    resync_period: 1h
+  k8s:
+    client:
+      timeout: 30000ms
+  authn:
```
Comment (polskikiel) [inline resources/service-manager-proxy/values.yaml]: why do we need that?

File: resources/service-manager-proxy/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+{{ toYaml . | indent 8 }}
+      {{- end }}
+      volumes:
+        - name: config-volume
+          configMap:
+            name: {{ template "service-broker-proxy.fullname" . }}-config
+      serviceAccountName: {{ template "service-broker-proxy.fullname" . }}
+      {{- if .Values.image.pullsecret }}
+      imagePullSecrets:
+      - name: {{.Values.image.pullsecret}}
```
Comment (polskikiel) [inline resources/service-manager-proxy/templates/deployment.yaml]: add spaces here between brackets

**Comment by mszostok** [comment]
@polskikiel the chart was copied from the upstream I do not want to change it because it will be problematic to do chart upgrades in future better is to have it in the same shape as in upstream. 

NOTE: I added only the annotation for not injecting the istio sidecar

**Comment by mszostok** [comment]
For more info about this topic please check comments under this issue: https://github.com/kyma-project/kyma/issues/6325

**Pull request merged**